File: src/dataset/javascript/largest-series-product_exercism-ThomasZumsteg/claude/ClaudeSonnet4_largest-series-product_v2.js
var Series = function(digits) {
	/* A series of digits */
	this.digits = digits;
	this.length = digits.length;
}

Series.prototype.slices = function(size) {
	/* Slices digits into groups */
	if(this.length < size)
		throw new Error("Slice size is too big.");
	var slices = [];
	for(var i = 0; i + size <= this.length; i++ ) {
		var slice = [];
		for(var j = 0; j < size; j++) {
			slice.push(parseInt(this.digits[i + j]));
		}
		slices.push(slice);
	}
	return slices;
};

Series.prototype.largestProduct = function(size) {
	/* The largest group when digits of slices are multipled together */
	if(this.length < size)
		throw new Error("Slice size is too big.");
	
	var maxProduct = 0;
	for(var i = 0; i + size <= this.length; i++) {
		var product = 1;
		for(var j = 0; j < size; j++) {
			product *= parseInt(this.digits[i + j]);
		}
		if(product > maxProduct) {
			maxProduct = product;
		}
	}
	return maxProduct;
};

function multiply(numbers) {
	/* Multiples all elements of an array together */
	var result = 1;
	for(var i = 0; i < numbers.length; i++) {
		result *= numbers[i];
	}
	return result;
}

module.exports = Series;

File: src/dataset/python/largest-series-product_exercism-fruit-in/claude/ClaudeSonnet4_largest_series_product_v2.py
def largest_product(series, size):
    if len(series) < size or size < 0 or any(not x.isdigit() for x in series):
        raise ValueError(r".+")

    if size == 0:
        return 1

    digits = [int(x) for x in series]
    max_product = 0

    for i in range(len(digits) - size + 1):
        if 0 in digits[i:i + size]:
            continue
        
        product = 1
        for j in range(i, i + size):
            product *= digits[j]
        
        if product > max_product:
            max_product = product

    return max_product